xbmc/video/VideoDatabase.cpp

```diff
@@ -5255,14 +5255,24 @@ bool CVideoDatabase::GetPlayCounts(const std::string &strPath, CFileItemList &it
       CFileItemPtr item = items.Get(path);
       if (item)
       {
-        item->GetVideoInfoTag()->SetPlayCount(m_pDS->fv(1).get_asInt());
+        if (!items.IsPlugin() || item->GetVideoInfoTag()->GetPlayCount() == -1)
+          item->GetVideoInfoTag()->SetPlayCount(m_pDS->fv(1).get_asInt());
+
         if (!item->GetVideoInfoTag()->GetResumePoint().IsSet())
         {
           item->GetVideoInfoTag()->SetResumePoint(m_pDS->fv(2).get_asInt(), m_pDS->fv(3).get_asInt());
         }
       }
       m_pDS->next();
     }
+
+    // if playcount is still -1 set it to zero
+    for (CFileItemPtr pItem : items)
```

Comment:
Yes but I'm not sure if it should be the responsibility of `CVideoDatabase` to handle these cases because it shouldn't know what to do with files that it doesn't know about. IMO it should only touch files that it is tracking.